admin/worker_4tests.m.template

```diff
@@ -13,14 +13,62 @@ function [ok,err_mess,je]=worker_4tests(worker_controls_string)
 %              contains only minimal initialization information, namely the
 %              folder name where the job initialization data are located on
 %              a remote system.
-
+je = [];
 if isempty(which('herbert_init.m'))
-    horace_on();
-    % Set up non-default herbert or Horace path to test them
-    % herbert_on('/home/isis_direct_soft/Herbert_bugfix/herbert_core');
-    % horace_on('/home/isis_direct_soft/Horace_bugfix/horace_core');
+    try
+        horace_on()
+    catch ME
+        ok = false;
+        err_mess = ME;
+        pid = int64(feature('getpid'));
+        log_file = sprintf('WORKER_V2_pid%i_PackageInit_failure.log',pid);
+        write_fail_log(log_file,err_mess)
+        return;
+    end
+end
+DO_LOGGING = false;
+DO_DEBUGGING = false;
+try
+    [ok, err_mess,je] = parallel_worker(worker_controls_string,DO_LOGGING,DO_DEBUGGING);
```

Comment:
there is a lot of logic in this catch block and the ~ok block afterwards; can this be simplified or extracted to a helper function?